devito/ir/support/basic.py

```diff
@@ -284,6 +284,13 @@ def is_irregular(self):
     def is_scalar(self):
         return self.rank == 0
 
+    def affine_if_present(self, findices):
+        """
+        Return True if the provided ``findices`` appear in ``self`` and are
+        affine, False otherwise.
+        """
```

Comment:
Docstring is a bit misleading, if I'm understanding the code correctly?

"Returns False if findices appear in self and are _not_ affine, True otherwise".